tests/inspectdb/tests.py

```diff
@@ -121,34 +119,34 @@ def test_number_field_types(self):
 
         assertFieldType('int_field', "models.IntegerField()")
 
-        if connection.features.can_introspect_positive_integer_field:
-            assertFieldType('pos_int_field', "models.PositiveIntegerField()")
-        else:
-            assertFieldType('pos_int_field', "models.IntegerField()")
-
-        if connection.features.can_introspect_positive_integer_field:
-            if connection.features.can_introspect_big_integer_field:
-                assertFieldType('pos_big_int_field', 'models.PositiveBigIntegerField()')
-            else:
-                assertFieldType('pos_big_int_field', 'models.PositiveIntegerField()')
-            if connection.features.can_introspect_small_integer_field:
-                assertFieldType('pos_small_int_field', "models.PositiveSmallIntegerField()")
-            else:
-                assertFieldType('pos_small_int_field', "models.PositiveIntegerField()")
-        else:
-            if connection.features.can_introspect_big_integer_field:
-                assertFieldType('pos_big_int_field', 'models.BigIntegerField()')
-            else:
-                assertFieldType('pos_big_int_field', 'models.IntegerField()')
-            if connection.features.can_introspect_small_integer_field:
-                assertFieldType('pos_small_int_field', "models.SmallIntegerField()")
-            else:
-                assertFieldType('pos_small_int_field', "models.IntegerField()")
-
-        if connection.features.can_introspect_small_integer_field:
-            assertFieldType('small_int_field', "models.SmallIntegerField()")
+        assertFieldType(
+            'pos_int_field',
+            'models.%s()' % connection.features.introspected_field_types['PositiveIntegerField']
+        )
+
+        if connection.features.introspected_field_types['PositiveIntegerField'] == 'PositiveIntegerField':
+            assertFieldType(
+                'pos_big_int_field',
+                'models.Positive%s()' % connection.features.introspected_field_types['BigIntegerField']
```

Comment:
I don't think the if statement is needed. Shouldn't this be `introspected_field_types['PositiveBigIntegerField']`?

Reply:
we don't have `PositiveBigIntegerField` in `introspected_field_types`

Reply:
Why not add it?

Reply:
`PositiveBigIntegerField` and `PositiveSmallIntegerField` added